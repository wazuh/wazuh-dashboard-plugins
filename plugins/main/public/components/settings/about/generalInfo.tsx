import {
  EuiButtonIcon,
  EuiFlexGroup,
  EuiFlexItem,
  EuiPanel,
  EuiSpacer,
  EuiText,
  EuiTitle,
} from '@elastic/eui';
import React from 'react';
import { getHttp } from '../../../kibana-services';
import { getAssetURL } from '../../../utils/assets';

interface SettingsAboutGeneralInfoProps {
  pluginAppName: string;
}

export const SettingsAboutGeneralInfo = ({
  pluginAppName,
}: SettingsAboutGeneralInfoProps) => {
  const googleGroupsSVG = getHttp().basePath.prepend(
    getAssetURL('images/icons/google_groups.svg'),
  );

  const Section = ({
    title,
    content,
  }: {
    title: string;
    content: React.ReactNode;
  }) => (
    <EuiFlexItem>
      <EuiPanel paddingSize='l'>
        <EuiTitle size='m'>
          <h2>{title}</h2>
        </EuiTitle>
        <EuiSpacer size='l' />
        {content}
      </EuiPanel>
    </EuiFlexItem>
  );

  return (
    <EuiFlexGroup gutterSize='l' direction='row' responsive>
      <Section
        title={`Welcome to the ${pluginAppName}`}
        content={
          <EuiText size='m'>
            <p>
              Dashboard provides management and monitoring capabilities, giving
              users control over the infrastructure. You can monitor your agents
              status and configuration, query and visualize your alert data and
              monitor manager rules and configuration.
            </p>
          </EuiText>
        }
      />
      <Section
        title='Community'
        content={
          <div>
            <EuiText size='m'>
              <p>
                Enjoy your experience and please don't hesitate to give us your
                feedback.
              </p>
            </EuiText>
            <EuiSpacer size='l' />
            <EuiFlexGroup
              alignItems='center'
              justifyContent='center'
              gutterSize='xl'
              responsive={false}
            >
              <EuiFlexItem grow={false}>
                <EuiButtonIcon
                  aria-label='Slack'
                  iconType='logoSlack'
                  iconSize='xxl'
                  href='https://wazuh.com/community/join-us-on-slack/'
                  target='_blank'
                >
                  Slack
                </EuiButtonIcon>
              </EuiFlexItem>
              <EuiFlexItem grow={false}>
                <EuiButtonIcon
                  aria-label='Google Groups'
                  iconType={googleGroupsSVG}
                  iconSize='xxl'
                  href='https://groups.google.com/forum/#!forum/wazuh'
                  target='_blank'
                >
                  Google groups
                </EuiButtonIcon>
              </EuiFlexItem>
              <EuiFlexItem grow={false}>
                <EuiButtonIcon
                  aria-label='Github'
                  iconType='logoGithub'
                  iconSize='xxl'
                  href='https://github.com/wazuh/wazuh-kibana-app'
                  target='_blank'
                >
                  Github
                </EuiButtonIcon>
              </EuiFlexItem>
            </EuiFlexGroup>
          </div>
        }
      />
    </EuiFlexGroup>
  );
};
