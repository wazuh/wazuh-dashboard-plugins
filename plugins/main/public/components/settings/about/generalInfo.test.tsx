import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { SettingsAboutGeneralInfo } from './generalInfo';

jest.mock('../../../kibana-services', () => ({
  getHttp: jest.fn().mockReturnValue({
    basePath: { prepend: (url: string) => '<svg></svg>' },
  }),
}));

jest.mock('../../../utils/assets', () => ({
  getAssetURL: jest.fn().mockReturnValue(null),
}));

describe('SettingsAboutGeneralInfo component', () => {
  test('should render a component', () => {
    const { container, getByText } = render(
      <SettingsAboutGeneralInfo pluginAppName="Wazuh dashboard" />
    );

    expect(container).toMatchSnapshot();

    const panels = container.getElementsByClassName('euiPanel');
    expect(panels.length).toBe(2);

    expect(getByText('Welcome to the Wazuh dashboard')).toBeInTheDocument();
    expect(getByText('Community')).toBeInTheDocument();
  });
});
