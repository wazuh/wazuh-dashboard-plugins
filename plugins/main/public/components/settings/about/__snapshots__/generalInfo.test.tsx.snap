// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsAboutGeneralInfo component should render a component 1`] = `
<div>
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      class="euiFlexItem"
    >
      <div
        class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
      >
        <h2
          class="euiTitle euiTitle--medium"
        >
          Welcome to the Wazuh dashboard
        </h2>
        <div
          class="euiSpacer euiSpacer--l"
        />
        <div
          class="euiText euiText--medium"
        >
          <p>
            Wazuh dashboard
             provides management and monitoring capabilities, giving users control over the Wazuh infrastructure. You can monitor your agents status and configuration, query and visualize your alert data and monitor manager rules and configuration.
          </p>
        </div>
      </div>
    </div>
    <div
      class="euiFlexItem"
    >
      <div
        class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
      >
        <h2
          class="euiTitle euiTitle--medium"
        >
          Community
        </h2>
        <div
          class="euiSpacer euiSpacer--l"
        />
        <div>
          <div
            class="euiText euiText--medium"
          >
            <p>
              Enjoy your Wazuh experience and please don't hesitate to give us your feedback.
            </p>
          </div>
          <div
            class="euiSpacer euiSpacer--l"
          />
          <div
            class="euiFlexGroup euiFlexGroup--gutterExtraLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentCenter euiFlexGroup--directionRow"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <a
                aria-label="Slack"
                class="euiButtonIcon euiButtonIcon--primary euiButtonIcon--empty euiButtonIcon--xSmall"
                href="https://wazuh.com/community/join-us-on-slack/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--xxLarge euiIcon--inherit euiIcon-isLoading euiButtonIcon__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </a>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <a
                aria-label="Google Groups"
                class="euiButtonIcon euiButtonIcon--primary euiButtonIcon--empty euiButtonIcon--xSmall"
                href="https://groups.google.com/forum/#!forum/wazuh"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  class="euiIcon euiIcon--xxLarge euiIcon--inherit euiButtonIcon__icon"
                  src="<svg></svg>"
                />
              </a>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <a
                aria-label="Github"
                class="euiButtonIcon euiButtonIcon--primary euiButtonIcon--empty euiButtonIcon--xSmall"
                href="https://github.com/wazuh/wazuh-kibana-app"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--xxLarge euiIcon--inherit euiIcon-isLoading euiButtonIcon__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
